presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/TransformUncorrelatedInPredicateSubqueryToSemiJoin.java

```diff
@@ -216,7 +216,8 @@ public PlanNode visitProject(ProjectNode node, RewriteContext<Void> context)
                     inPredicate,
                     mapAssignmentSymbolsToExpression(node.getAssignments()));
 
-            ProjectNode rewrittenNode = (ProjectNode) context.rewrite(node);
+            // cannot use context.rewrite() as it ends with endless loop
```

Comment:
No, it should use context.defaultRewrite. That will cause the rewriter to call the default rewrite logic for the project node (i.e., skip the current class' implementation).